File: mobile/lib/features/markers/domain/entities/category.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'category.freezed.dart';

@freezed
class Category with _$Category {
  const factory Category({
    required String id,
    required String tripId,
    required String name,
    required String color,
    required DateTime createdAt,
  }) = _Category;
}

File: backend/domain/category.go
package domain

import (
	"context"
	"time"
)

type Category struct {
	ID        string    `json:"id"`
	TripID    string    `json:"trip_id"`
	Name      string    `json:"name"`
	Color     string    `json:"color"`
	CreatedAt time.Time `json:"created_at"`
}

type CategoryRepository interface {
	GetCategories(ctx context.Context, token, tripID string) ([]Category, error)
}

type CategoryUsecase interface {
	GetCategories(ctx context.Context, token, tripID string) ([]Category, error)
}
